lib/mix/tasks/eunit.ex

```diff
@@ -36,25 +36,36 @@ defmodule Mix.Tasks.Eunit do
 
   """
 
+  @cover [output: "cover", tool: Mix.Tasks.Test.Cover]
+
   def run(args) do
     options = parse_options(args)
+    project = Mix.Project.config
 
     # add test directory to compile paths and add
     # compiler options for test
-    post_config = eunit_post_config(Mix.Project.config)
+    post_config = eunit_post_config(project)
     modify_project_config(post_config)
 
     # make sure mix will let us run compile
     ensure_compile
     Mix.Task.run "compile"
 
+    # start cover
+    cover =
+      if options[:cover] do
+        compile_path = Mix.Project.compile_path(project)
+        cover = Keyword.merge(@cover, project[:test_coverage] || [])
+        cover[:tool].start(compile_path, cover)
+      end
+
     # run the actual tests
-    if(options[:cover], do: cover_start())
     test_modules(post_config[:erlc_paths], options[:patterns])
     |> Enum.map(&module_name_from_path/1)
     |> Enum.drop_while(fn(m) ->
       tests_pass?(m, options[:eunit_opts] ++ post_config[:eunit_opts]) end)
-    if(options[:cover], do: cover_analyse())
+
+    cover && cover.()
```

Comment:
`cover` is either `nil` or a function. `Cover.start/2` returns a function that does the analysis and writes the output.

The code for the default `Cover` module is here: https://github.com/elixir-lang/elixir/blob/master/lib/mix/lib/mix/tasks/test.ex#L2